### DISCUSSION TITLE
[Closed] Node 16 support

### DISCUSSION THREAD

**Discussion by milanzigmond** [meteor/meteor#12015] [Feature Requests]
For some packages like opensea-js the node 16 is required to make sure common crypto dependencies work. Any plans to support node 16 in the near future? Or is there any way how I can update node version for my project? Thank you

<img width="1393" alt="Screenshot 2022-04-23 at 15 26 43" src="https://user-images.githubusercontent.com/4190850/164896559-cb5adfc7-ae98-49e6-8b96-af6fe8c8af8a.png">

<img width="773" alt="Screenshot 2022-04-23 at 15 27 15" src="https://user-images.githubusercontent.com/4190850/164896584-ea11e967-af95-4058-88b6-2c1329bbf172.png">

**Comment by StorytellerCZ** [comment]
Work is underway to upgrade to Node 16. Fibers are the blocker for this upgrade.
Replacing them is a major undertaking and is being documented here:
https://github.com/meteor/meteor/discussions/11505

There is no way for you to update the Meteor version of Node for your app. Point of Meteor is that it is [pinned to a specific version of Node](https://blog.meteor.com/know-your-node-js-ff0b3a0f634e).

**Reply by milanzigmond** [reply]
I see, thank you Jan. I hope you guys solve it fast, this is starting to be cumbersome to use meteor with any popular library. I'll read the Discussion and see what I can do to help.

**Comment by diavrank** [comment]
[Node 18](https://medium.com/@Luna-Rojas/nodejs-18-is-here-3-features-that-will-blow-your-mind-7c2b86e1d13) is here. I don't want to press but is there an approximate date when Meteor is updated with Node 16?

**Reply by StorytellerCZ** [reply]
2.8 is the first major step towards it (`*Async` added to function).
In my opinion Meteor 3.0 should be with Node 16 (or jump straight to 18) with everything using async/await as the Fibers will stop working once updated to Node 16.

**Reply by diavrank** [reply]
Hi @StorytellerCZ , is there an ETA for the release of Meteor 3.0 (with Node 16+)?

**Reply by StorytellerCZ** [reply]
I don't know. @Grubba27 probably has the best idea on that.

**Comment by foyzulkarim** [comment]
[Node 19](https://nodejs.org/en/blog/announcements/v19-release-announce/) is here and we have less than 6 months to migrate to 16.

Quote:
> Also of note is that Node.js 14 will go End-of-Life in April 2023, so we advise you to start planning to upgrade to Node.js 16 (LTS) or Node.js 18 (soon to be LTS).
> Please, consider that Node.js 16 (LTS) will go End-of-Life in September 2023

**Reply by Julusian** [reply]
Follow the progress on migrating away from 14 in https://github.com/meteor/meteor/discussions/11505

Personally, I do not think it will be done before the EOL of 14, unless there is a lot of non-public work in progress

**Reply by sherryysj** [reply]
LOL, nice to see you here Foyzul! I am trying to use `OpenAI` on meteor project and it requires `AbortController`, which requires node 16, so I am googling answers to solve it and it takes me here, LOL.

**Reply by awatson1978** [reply]
I just found my way here because I was looking at OpenAI libraries also.   NodeOnFHIR stack has mostly migrated to await/async.  Looking forward to Meteor 3.0.

**Comment by cormip** [comment]
If migrating up from Node v14 is not imminent, perhaps it'd be a good idea to maintain a list of modern npm packages that aren't compatible to avoid us developers wasting our time before discovering a package isn't compatible with Meteor.

Add [@simplewebauthn/server](https://github.com/MasterKale/SimpleWebAuthn/tree/master/packages/server) to the list because of its reliance on the SubtleCrypto class available in Node v15+.

**Reply by rj-david** [reply]
Instead of creating a list, scan the changelog/releases of the package before installing. Most packages that maintain changelogs indicate if they are dropping support for Node 14.

e.g. for simplewebauthn: "In addition, the library now targets Node 16 and above"

**Reply by StorytellerCZ** [reply]
You can also just check `package.json` where they often show the `engine` as well. Maintaining a list of packages that are no longer compatible with Node 14 is not feasible give that there is over two millions of them. Also the work towards Meteor 3.0 is going according to schedule.

**Reply by cormip** [reply]
Yeah, good call on scanning `package.json`. I checked one of my apps and all packages but one specified the supported node versions. It's not helping sell MeteorJS as a viable platform to my tech-savvy clients when they realize it depends on a Node version that is 15 months past Active Support and 3 months away from EoL on Security Support. Go Meteor 3.0!!!

**Comment by markpdev2** [comment]
Thanks is there any progress on this its basically end of life at the moment

**Reply by StorytellerCZ** [reply]
Status updates here https://github.com/meteor/meteor/discussions/11505 and on the Meteor forums.

**Reply by hsinewu** [reply]
wow, want to try this out and noticed the version is not supported
node has LTS v18 and meteor not even support v16
time to question my entire life

**Reply by StorytellerCZ** [reply]
Meteor v3 is going to run on Node 18 for a bit until it will upgrade to Node 20 (my personal guess is that is going to happen in few feature releases).

**Comment by fredmaiaarantes** [comment]
To update this discussion, Meteor 3.0 is on its way and will run on Node.js 20. A few beta versions were released, and we encourage everyone to test them and give us feedback. For more information, check out the ongoing discussion below. 

> What's left until an official Meteor 3.0?
> https://github.com/meteor/meteor/discussions/12865

**Comment by fredmaiaarantes** [comment]
Meteor 3.0 is here with Node.js 20! 🎉 
https://dev.to/meteor/meteor-3-is-officially-here-3gml
